source-container-build/app/source_build.py

```diff
@@ -507,7 +536,29 @@ def download_parent_image_sources(source_image: str, work_dir: str) -> str:
     sources_dir = create_dir(work_dir, "parent_image_sources")
     logger.info("Copy source image %s into directory %s", source_image, sources_dir)
     # skopeo can not copy signatures to oci image layout
-    skopeo_copy(f"docker://{source_image}", f"oci:{sources_dir}", remove_signatures=True)
+
+    try:
+        skopeo_copy(
+            f"docker://{source_image}",
+            f"oci:{sources_dir}",
+            remove_signatures=True,
+        )
+    except NoSignatureError:
+        logger.info(
+            "Source image has no signature, deleting:",
+            os.path.join(
+                work_dir,
+                "parent_image_sources",
+            ),
+        )
+        shutil.rmtree(os.path.join(work_dir, "parent_image_sources"))
+        if ignore_unsigned_image:
+            logger.info("Ignored missing source image signatures")
+            return ""
+
+        logger.warning("Source image rejected: A signature was required, but no signature exists")
+        raise
```

Comment:
Rather than log a hardcoded warning message, IMO we should use the stderr of the failed skopeo process
```suggestion
        raise
```
(tied to the next suggestion)

Reply:
Thanks for the suggestion!